src/service_provider/service_provider.routes.spec.js

```diff
@@ -0,0 +1,53 @@
+import request from "supertest";
+import app from "../app";
+import { FindAllServiceProviders, CreateServiceProvider, FindAllServiceProvidersById, UpdateServiceProvider, DeleteServiceProvider } from './service_provider.services'
+
+jest.mock('./service_provider.services')
+
+const fakeServiceProviders = [
+    {
+        id: 'aaabbb111',
+        name: 'test'
+    }
+]
+
+describe("Service Provider Router Tests", () => {
+  test("GET /service_provider should return empty array", done => {
+      FindAllServiceProviders.mockImplementation(() => fakeServiceProviders)
+      request(app)
+        .get("/service_provider")
+        .then(response => {
+          expect(response.statusCode).toBe(200);
+          expect(response.body).toStrictEqual(fakeServiceProviders)
+          done();
+        });
+    });
+
+  test("GET /service_provider/:id should return empty array", done => {
+      FindAllServiceProvidersById.mockImplementation(() => fakeServiceProviders[0])
+      request(app)
+        .get("/service_provider/1")
+        .then(response => {
+          expect(response.statusCode).toBe(200);
+          expect(response.body).toStrictEqual(fakeServiceProviders[0])
+          done();
+        });
+    });
+
+    test("Post /service_provider/:id should return empty array", done => {
+        const mockCreateData = {
+          id: '10',
+          name: 'test'
+        }
+        CreateServiceProvider.mockImplementation((args) => mockCreateData)
+        request(app)
+          .post("/service_provider")
+          .send(mockCreateData)
+          .set('Accept', 'application/json')
+          .then(response => {
+            expect(response.statusCode).toBe(200);
+            expect(response.body).toStrictEqual(mockCreateData)
+            done();
+          });
+    });
+});
```

Comment:
These seem to fall under the category of "functional tests" right? Testing the PUT and DELETE routes seems like they would need to fire up some dummy db with some seeded data? Can these be configured with some global `beforeAll()` to fire up some database, or have a separate script do that before running the tests in the `npm test` script?

Reply:
Yeah these are all just unit tests. They seem simple only because these routes don't have a lot of business logic yet.

I'm not testing whether or not the db returns the right stuff, etc etc